OpenRA.Mods.Common/Activities/Demolish.cs

```diff
@@ -57,6 +58,14 @@ protected override void OnInside(Actor self)
 				if (cloak != null && cloak.Info.UncloakOn.HasFlag(UncloakType.Demolish))
 					cloak.Uncloak();
 
+				if (!chargePlanted)
+				{
+					chargePlanted = true;
```

Comment:
There couléd be multiple actors entering to the same target (D2k Saboteurs came to my mind) in which case multiple `Lock`s might be errandeous.